app/controllers/events_controller.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+class EventsController < ApplicationController
+  before_action :signed_in?, :set_event, only: %i[show edit update]
```

Comment:
before_action is not enough.
You should redirect at before action if not signed in.